gensim/matutils.py

```diff
@@ -426,7 +426,7 @@ def unitvec(vec, norm='l2'):
         if norm == 'l2':
             veclen = blas_nrm2(vec)
         if veclen > 0.0:
-            return blas_scal(1.0 / veclen, vec)
+            return blas_scal(1.0 / veclen, vec).astype(vec.dtype)
```

Comment:
Hi @accraze , thanks for the fix! A better solution here might be to modify the hardcoded dtype in line 423 above, it simplifies the logic, and also ensures that the dtype is consistent for vectors with all zeros too (a rather trivial and probably uncommon case, of course)

Reply:
Hey @jayantj, I looked into this, however `blas_scal` returns an array of type `float` (see line 398). Not sure if there is a better way to handle this...

Reply:
@jayantj What else needs to done? `blas_scal` is not being used anywhere. So, should I define `blas_scal` before line 429 and remove hardcoded float from the definition of `blas_scal`